src/ngraph_mark_for_clustering.cc

```diff
@@ -266,6 +266,7 @@ Status MarkForClustering(Graph* graph) {
       type_constraint_map["Shape"]["T"] = NGraphDTypes();
       type_constraint_map["Shape"]["out_type"] = NGraphIndexDTypes();
       type_constraint_map["Sigmoid"]["T"] = NGraphNumericDTypes();
+      type_constraint_map["SigmoidGrad"]["T"] = NGraphNumericDTypes();
       type_constraint_map["Sign"]["T"] = NGraphNumericDTypes();
```

Comment:
Not sure if SigmoidGrad would work or make sense for int types. Can you please double check what `T` ought to be

If it works for int types, lets make one of the test cases with int numbers

Reply:
I verified in the gen_math_ops.py file, the input to SigmoidGrad only supports `bfloat16`, `half`, `float32`, `float64`, `complex64`, `complex128`.

So there is no int type, then I need to change to `NgraphDTypes()` for non-int type constraints?

Reply:
For most ops, we are using NGraphNumericDTypes() as a superset, expecting tf to complain about wrong datatypes, before it reaches us. Either tf will complain, or tf (without nGraph) would fail too, worth checking.

Reply:
I think we should have "DT_FLOAT, DT_DOUBLE" in a separate ArraySlice, but that is a change not related to this PR I suppose. So no NGraphNumericDTypes() looks fine for now.

Reply:
Sounds good. I will still use `NGraphNumericDTypes()` for SigmoidGrad op.